src/fork_gc.c

```diff
@@ -290,7 +290,12 @@ static void sendHeaderString(ForkGC *gc, void *arg) {
 }
 
 static void FGC_reportProgress(ForkGC *gc) {
+#if defined(COV) || defined(COVERAGE)
+  // Don't send heartbeats when running coverage tests, as it can mess with the coverage report.
+  return;
+#else
   RedisModule_SendChildHeartbeat(gc->progress);
+#endif
```

Comment:
Consider simplifying the logic (no `else` case)
```suggestion
#if !defined(COV) && !defined(COVERAGE)
  // Don't send heartbeats when running coverage tests, as it can mess with the coverage report.
  RedisModule_SendChildHeartbeat(gc->progress);
#endif
```